### ISSUE TITLE
[JENKINS-54512] can not change workspace in k8s pipeline

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/kubernetes-plugin#2122]
<p>I am a developer of the go language, running tests using the jenkins k8s agent. I need to put my workspace into GOPATH. But when I switch the working directory using `ws / dir`, I get the following error.</p>

<p> </p>

<p> </p>

<p> </p>

<pre>
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from 10.244.5.0/10.244.5.0:58310
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1741)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:357)
		at hudson.remoting.Channel.call(Channel.java:955)
		at hudson.FilePath.act(FilePath.java:1071)
		at hudson.FilePath.act(FilePath.java:1060)
		at hudson.FilePath.mkdirs(FilePath.java:1245)
		at org.jenkinsci.plugins.durabletask.FileMonitoringTask$FileMonitoringController.<init>(FileMonitoringTask.java:171)
		at org.jenkinsci.plugins.durabletask.BourneShellScript$ShellController.<init>(BourneShellScript.java:197)
		at org.jenkinsci.plugins.durabletask.BourneShellScript$ShellController.<init>(BourneShellScript.java:189)
		at org.jenkinsci.plugins.durabletask.BourneShellScript.launchWithCookie(BourneShellScript.java:110)
		at org.jenkinsci.plugins.durabletask.FileMonitoringTask.launch(FileMonitoringTask.java:98)
		at org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.start(DurableTaskStep.java:264)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:270)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:180)
		at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:122)
		at sun.reflect.GeneratedMethodAccessor616.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
		at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
		at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
		at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
		at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
		at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
		at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
		at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:157)
		at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
		at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:155)
		at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:159)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:129)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:129)
		at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
		at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
		at sun.reflect.GeneratedMethodAccessor497.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
		at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
		at com.cloudbees.groovy.cps.Next.step(Next.java:83)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
		at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:122)
		at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:261)
		at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$101(SandboxContinuable.java:34)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.lambda$run0$0(SandboxContinuable.java:59)
		at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:58)
		at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:174)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:332)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$200(CpsThreadGroup.java:83)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:244)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:232)
		at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
		at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
		at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.nio.file.AccessDeniedException: /go
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
	at java.nio.file.Files.createDirectory(Files.java:674)
	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
	at java.nio.file.Files.createDirectories(Files.java:767)
	at hudson.FilePath.mkdirs(FilePath.java:3272)
	at hudson.FilePath.access$1300(FilePath.java:212)
	at hudson.FilePath$Mkdirs.invoke(FilePath.java:1253)
	at hudson.FilePath$Mkdirs.invoke(FilePath.java:1249)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3085)
	at hudson.remoting.UserRequest.perform(UserRequest.java:212)
	at hudson.remoting.UserRequest.perform(UserRequest.java:54)
	at hudson.remoting.Request$2.run(Request.java:369)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:93)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748)
Finished: FAILURE
</pre>
<p> </p>

<p>my jenkinsfile is </p>

<p> </p>

<pre>
pipeline {
  agent {
    kubernetes {
      label <span class="code-quote">'golang-sample'</span>
      yaml '''
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: docker-builder
spec:
  containers:
  - name: golang
    image: golang:1.10-alpine
    command:
    - cat
    tty: <span class="code-keyword">true</span>
'''
      defaultContainer <span class="code-quote">'golang'</span>
    }

  }
  stages {
		stage(<span class="code-quote">'test'</span>){
			steps{
				dir(<span class="code-quote">'/go/src/github.com/runzexia/kubesphere-devops-sample-bin'</span>){
				    container(<span class="code-quote">'golang'</span>){
					sh <span class="code-quote">'pwd && ls -a'</span>
				    }
				}
			}
		}
  }
}
</pre>

<pre>
pipeline {
  agent {
    kubernetes {
      label <span class="code-quote">'golang-sample'</span>
      yaml '''
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: docker-builder
spec:
  containers:
  - name: golang
    image: golang:1.10-alpine
    command:
    - cat
    tty: <span class="code-keyword">true</span>
'''
      defaultContainer <span class="code-quote">'golang'</span>
    }

  }
  stages {
		stage(<span class="code-quote">'test'</span>){
			steps{
				ws(<span class="code-quote">'/go/src/github.com/runzexia/kubesphere-devops-sample-bin'</span>){
				    container(<span class="code-quote">'golang'</span>){
					sh <span class="code-quote">'pwd && ls -a'</span>
				    }
				}
			}
		}
  }
}
</pre>
<p> </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=runzexia">runzexia</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512" target="_blank">can not change workspace in k8s pipeline</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: kubernetes-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2021-01-08T14:18:37+00:00
<li><b>votes</b>: 3
<li><b>watchers</b>: 12
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I am a developer of the go language, running tests using the jenkins k8s agent. I need to put my workspace into GOPATH. But when I switch the working directory using `ws / dir`, I get the following error.</p>

<p> </p>

<p> </p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from 10.244.5.0/10.244.5.0:58310
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1741)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:357)
		at hudson.remoting.Channel.call(Channel.java:955)
		at hudson.FilePath.act(FilePath.java:1071)
		at hudson.FilePath.act(FilePath.java:1060)
		at hudson.FilePath.mkdirs(FilePath.java:1245)
		at org.jenkinsci.plugins.durabletask.FileMonitoringTask$FileMonitoringController.&lt;init&gt;(FileMonitoringTask.java:171)
		at org.jenkinsci.plugins.durabletask.BourneShellScript$ShellController.&lt;init&gt;(BourneShellScript.java:197)
		at org.jenkinsci.plugins.durabletask.BourneShellScript$ShellController.&lt;init&gt;(BourneShellScript.java:189)
		at org.jenkinsci.plugins.durabletask.BourneShellScript.launchWithCookie(BourneShellScript.java:110)
		at org.jenkinsci.plugins.durabletask.FileMonitoringTask.launch(FileMonitoringTask.java:98)
		at org.jenkinsci.plugins.workflow.steps.durable_task.DurableTaskStep$Execution.start(DurableTaskStep.java:264)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:270)
		at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:180)
		at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:122)
		at sun.reflect.GeneratedMethodAccessor616.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
		at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
		at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
		at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
		at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
		at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
		at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
		at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:157)
		at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
		at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:155)
		at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:155)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:159)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:129)
		at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:129)
		at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
		at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:57)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:109)
		at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixArg(FunctionCallBlock.java:82)
		at sun.reflect.GeneratedMethodAccessor497.invoke(Unknown Source)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
		at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
		at com.cloudbees.groovy.cps.Next.step(Next.java:83)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
		at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
		at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:122)
		at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:261)
		at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$101(SandboxContinuable.java:34)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.lambda$run0$0(SandboxContinuable.java:59)
		at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:108)
		at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:58)
		at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:174)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:332)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$200(CpsThreadGroup.java:83)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:244)
		at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:232)
		at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:64)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
		at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
		at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.nio.file.AccessDeniedException: /go
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
	at java.nio.file.Files.createDirectory(Files.java:674)
	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
	at java.nio.file.Files.createDirectories(Files.java:767)
	at hudson.FilePath.mkdirs(FilePath.java:3272)
	at hudson.FilePath.access$1300(FilePath.java:212)
	at hudson.FilePath$Mkdirs.invoke(FilePath.java:1253)
	at hudson.FilePath$Mkdirs.invoke(FilePath.java:1249)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3085)
	at hudson.remoting.UserRequest.perform(UserRequest.java:212)
	at hudson.remoting.UserRequest.perform(UserRequest.java:54)
	at hudson.remoting.Request$2.run(Request.java:369)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:93)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748)
Finished: FAILURE
</pre>
</div></div>
<p> </p>

<p>my jenkinsfile is </p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">pipeline {
  agent {
    kubernetes {
      label <span class="code-quote">'golang-sample'</span>
      yaml '''
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: docker-builder
spec:
  containers:
  - name: golang
    image: golang:1.10-alpine
    command:
    - cat
    tty: <span class="code-keyword">true</span>
'''
      defaultContainer <span class="code-quote">'golang'</span>
    }

  }
  stages {
		stage(<span class="code-quote">'test'</span>){
			steps{
				dir(<span class="code-quote">'/go/src/github.com/runzexia/kubesphere-devops-sample-bin'</span>){
				    container(<span class="code-quote">'golang'</span>){
					sh <span class="code-quote">'pwd &amp;&amp; ls -a'</span>
				    }
				}
			}
		}
  }
}
</pre>
</div></div>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">pipeline {
  agent {
    kubernetes {
      label <span class="code-quote">'golang-sample'</span>
      yaml '''
apiVersion: v1
kind: Pod
metadata:
  labels:
    app: docker-builder
spec:
  containers:
  - name: golang
    image: golang:1.10-alpine
    command:
    - cat
    tty: <span class="code-keyword">true</span>
'''
      defaultContainer <span class="code-quote">'golang'</span>
    }

  }
  stages {
		stage(<span class="code-quote">'test'</span>){
			steps{
				ws(<span class="code-quote">'/go/src/github.com/runzexia/kubesphere-devops-sample-bin'</span>){
				    container(<span class="code-quote">'golang'</span>){
					sh <span class="code-quote">'pwd &amp;&amp; ls -a'</span>
				    }
				}
			}
		}
  }
}
</pre>
</div></div>
<p> </p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>runzexia:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=353356&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353356">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I found the problem, I can switch to any directory under `/home/jenkins`, but I can't switch to other directories outside. Is it because the user who runs the jenkins master?</p>
</pre>
</details>

</ul>
</details>

<p>I found the problem, I can switch to any directory under `/home/jenkins`, but I can't switch to other directories outside. Is it because the user who runs the jenkins master?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dcurrie.png" title="dcurrie's avatar" /> dcurrie:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=353379&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353379">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Why not just change your GOPATH then?</p>
</pre>
</details>

</ul>
</details>

<p>Why not just change your GOPATH then?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>runzexia:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=353380&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-353380">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Changing the GOPATH is working, but it would be better if you could change the working directory at will in the k8s agent.</p>
</pre>
</details>

</ul>
</details>

<p>Changing the GOPATH is working, but it would be better if you could change the working directory at will in the k8s agent.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>abergmeier:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=354030&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-354030">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I tried changing to `/var/jenkins` with the same error message.</p>
</pre>
</details>

</ul>
</details>

<p>I tried changing to `/var/jenkins` with the same error message.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>abergmeier:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=354042&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-354042">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Also note, that if `/home/jenkins` is mounted noexec (e.g. in OpenShift) that you cannot execute any script there.</p>

<p>And since this bug is present, there is no workaround for this!</p>
</pre>
</details>

</ul>
</details>

<p>Also note, that if `/home/jenkins` is mounted noexec (e.g. in OpenShift) that you cannot execute any script there.</p>

<p>And since this bug is present, there is no workaround for this!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/csanchez.png" title="csanchez's avatar" /> csanchez:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=355499&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-355499">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>have you tried with <tt>dir</tt> inside <tt>container</tt> ?</p>
</pre>
</details>

</ul>
</details>

<p>have you tried with <tt>dir</tt> inside <tt>container</tt> ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>runzexia:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=355559&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-355559">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=csanchez" class="user-hover" rel="csanchez">csanchez</a> yes i have tried with `dir`, It also ditn't work properly, there are similar errors</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=csanchez" class="user-hover" rel="csanchez">csanchez</a> yes i have tried with `dir`, It also ditn't work properly, there are similar errors</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>azelezni:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=357829&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-357829">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Having the same issue, I'm running some stages in a container with root user, but getting access denied when trying to change directories, even with securityContext set as bellow:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">spec:
 securityContext:
   runAsUser: 0
   fsGroup: 0
   privileged: <span class="code-keyword">true</span></pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Having the same issue, I'm running some stages in a container with root user, but getting access denied when trying to change directories, even with securityContext set as bellow:</p>

<pre>
spec:
 securityContext:
   runAsUser: 0
   fsGroup: 0
   privileged: <span class="code-keyword">true</span></pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mog:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=373237&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-373237">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm having same issue</p>

<p>ws() {} doesn't work with kubernets plugin.</p>

<p>customWorkspace throws AccessDeniedException</p>

<p>Does the directory need to be created in agent docker image or something? I'm working around it by mv:ing files into workspace folder.</p>
</pre>
</details>

</ul>
</details>

<p>I'm having same issue</p>

<p>ws() {} doesn't work with kubernets plugin.</p>

<p>customWorkspace throws AccessDeniedException</p>

<p>Does the directory need to be created in agent docker image or something? I'm working around it by mv:ing files into workspace folder.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>caseteroconamor:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=399704&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-399704">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm having the same issue as well, I just need to share a folder between 3-4 slaves working in parallel and writing to the same EFS volume at the same time and I don't seem to make it work <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>I'm having the same issue as well, I just need to share a folder between 3-4 slaves working in parallel and writing to the same EFS volume at the same time and I don't seem to make it work <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vlatombe.png" title="vlatombe's avatar" /> vlatombe:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=403144&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-403144">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The problem with the dir step is that it creates the directory using remoting, which always targets the agent container. It means this step can only work with a path that is shared between the different containers in the pod, so under the workspace essentially.</p>
</pre>
</details>

</ul>
</details>

<p>The problem with the dir step is that it creates the directory using remoting, which always targets the agent container. It means this step can only work with a path that is shared between the different containers in the pod, so under the workspace essentially.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vlatombe.png" title="vlatombe's avatar" /> vlatombe:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-54512?focusedId=403145&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-403145">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Won't fix due to technical limitations.</p>
</pre>
</details>

</ul>
</details>

<p>Won't fix due to technical limitations.</p>
